### PR TITLE
Added debugger support

### PR DESCRIPTION
**Pull request by jacobdunefsky**
Added debugger support to runtime

### CODE REVIEW

File: core/internal/runtime/repl.go [hunk 1]
```diff
@@ -106,16 +107,41 @@ func invokeServiceEndpoint(ctx context.Context, args []string) (exitCode int) {
...
+	Port int `json:"port"`
+}
+
 func getInformation(ctx context.Context, args []string) (exitCode int) {
 	option := strings.ToLower(config.GetSystemComponent)
 	var str string
 	var err error
 	switch option {
 	case "sidecar", "sidecars":
+		doCall := func(sidecar string) (addrTuple_t, error) {
```
[the remark this answers is not in the document]
Comment (jacobdunefsky) [reply core/internal/runtime/repl.go:123]: The former. Potentially, we could change GetTopology to have it return more information, but this seemed easier.

File: core/internal/runtime/runtime-main.go [hunk 2]
```diff
@@ -112,6 +114,16 @@ func server(listener net.Listener) http.Server {
...
 
+	// debugger
+	router.GET(base+"/debug/register", routeImplRegisterDebugger)
+	router.POST(base+"/debug/register", routeImplRegisterDebugger)
+	router.PUT(base+"/debug/register", routeImplRegisterDebugger)
+	/*router.POST(base+"/debug/breakpoint", routeImplSetBreakpoint)
+	router.DELETE(base+"/debug/breakpoint", routeImplUnsetBreakpoint)
+	router.DELETE(base+"/debug/unregister", routeImplUnregisterDebugger)
+	router.POST(base+"/debug/pause", routeImplPause)
+	router.DELETE(base+"/debug/pause", routeImplUnpause)*/
```
[the remark this answers is not in the document]
Comment (jacobdunefsky) [reply core/internal/runtime/runtime-main.go:125]: Not needed after the change to websockets. Nice catch.

File: core/internal/runtime/runtime-main.go [hunk 3]
```diff
@@ -145,6 +157,7 @@ func Main() {
 
 	var listenHost string
 	if config.KubernetesMode {
+		fmt.Println("Kubernetes mode.")
```
[the remark this answers is not in the document]
Comment (jacobdunefsky) [reply core/internal/runtime/runtime-main.go:160]: Yes, I should've caught that. Cleaned up.

File: core/internal/runtime/commands.go [hunk 4]
```diff
@@ -25,17 +25,99 @@ import (
...
 
 	"github.com/IBM/kar/core/internal/config"
 	"github.com/IBM/kar/core/pkg/logger"
 	"github.com/IBM/kar/core/pkg/rpc"
 	"github.com/IBM/kar/core/pkg/store"
 	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
 )
 
+type actorTuple_t struct {
```
[the remark this answers is not in the document]
Comment (jacobdunefsky) [reply core/internal/runtime/commands.go:39]: Done.

File: core/internal/runtime/commands.go [hunk 5]
```diff
@@ -131,11 +213,48 @@ func CallActor(ctx context.Context, actor Actor, path, payload, flow string, par
...
 		"path":    path,
 		"payload": payload}
+
 	bytes, err := json.Marshal(msg)
 	if err != nil {
 		return nil, err
 	} else {
-		bytes, err = rpc.Call(ctx, rpc.Destination{Target: rpc.Session{Name: actor.Type, ID: actor.ID, Flow: flow}, Method: actorEndpoint}, defaultTimeout(), parentID, bytes)
+		//bytes, err = rpc.Call(ctx, rpc.Destination{Target: rpc.Session{Name: actor.Type, ID: actor.ID, Flow: flow}, Method: actorEndpoint}, defaultTimeout(), parentID, bytes)
+
```
[the remark this answers is not in the document]
Comment (jacobdunefsky) [reply core/internal/runtime/commands.go:223]: Done.

File: core/internal/runtime/repl.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgrove-oss) [reply core/internal/runtime/repl.go:123]: ok.  Let me see take a quick cut at extending GetToplogy.  It crosses a bunch of layers, but I think is doable.

**Pull request merged**